evalml/tests/pipeline_tests/test_pipelines.py

```diff
@@ -2754,3 +2755,80 @@ def test_pipeline_transform_with_final_estimator(
         ),
     ):
         pipeline.transform(X, y)
+
+
+@patch("evalml.pipelines.components.LogisticRegressionClassifier.fit")
+def test_training_only_component_in_pipeline_fit(mock_fit):
+    X = pd.DataFrame(
+        {
+            "a": [i for i in range(9)] + [np.nan],
+            "b": [i % 3 for i in range(10)],
+            "c": [i % 7 for i in range(10)],
+        }
+    )
+    y = pd.Series([0] * 5 + [1] * 5)
+    pipeline = BinaryClassificationPipeline(
+        {
+            "Imputer": ["Imputer", "X", "y"],
+            "Drop Rows Transformer": [DropRowsTransformer, "Imputer.x", "y"],
+            "Logistic Regression Classifier": [
+                "Logistic Regression Classifier",
+                "Drop Rows Transformer.x",
+                "Drop Rows Transformer.y",
+            ],
+        },
+        parameters={"Drop Rows Transformer": {"indices_to_drop": [0, 9]}},
+    )
+    pipeline.fit(X, y)
+    assert len(mock_fit.call_args[0][0]) == 8
+
+
+def test_training_only_component_in_pipeline_predict():
+    # Test that calling predict() will not evaluate any training-only transformations
+    X = pd.DataFrame(
+        {
+            "a": [i for i in range(9)] + [np.nan],
+            "b": [i % 3 for i in range(10)],
+            "c": [i % 7 for i in range(10)],
+        }
+    )
+    y = pd.Series([0] * 5 + [1] * 5)
+    pipeline = BinaryClassificationPipeline(
+        {
+            "Imputer": ["Imputer", "X", "y"],
+            "Drop Rows Transformer": [DropRowsTransformer, "Imputer.x", "y"],
+            "Logistic Regression Classifier": [
+                "Logistic Regression Classifier",
+                "Drop Rows Transformer.x",
+                "Drop Rows Transformer.y",
+            ],
+        },
+        parameters={"Drop Rows Transformer": {"indices_to_drop": [9]}},
+    )
+    pipeline.fit(X, y)
+    preds = pipeline.predict(X)
+    assert len(preds) == 10
```

Comment:
Is the jist here that no rows have been dropped during prediction since DropRowsTransformer is `evaluate_training=True`?

Reply:
Yup!
![](https://media4.giphy.com/media/xTiTnG7GYgnEejzOdW/giphy.gif)